docs/tailwind-best-practices.md

```diff
@@ -0,0 +1,355 @@
+# Tailwind CSS Best Practices
+
+A comprehensive guide for using Tailwind effectively and consistently across our design system monorepo and engineering organization.
+
+## 🎯 Core Principles
+
+### 1. Component-First Approach
+
+- **Use Design System Components**: Always prefer our design system components over raw JSX elements with Tailwind classes
+- **Props Over Classes**: Prefer component props to control variants and styles, but use `className`/`twClassname`/`tw` when no equivalent prop exists
+- **Example**:
+
+  **React Web:**
+
+  ```tsx
+  // ❌ Don't
+  <div className="rounded-lg bg-alternative flex-1 p-4">
+    <span className="text-default text-s-body-md font-medium">Content</span>
+  </div>
+
+  // ✅ Do
+  <Box
+    borderRadius={BoxBorderRadius.Lg}
+    backgroundColor={BoxBackgroundColor.BackgroundAlternative}
+    p={4}
+    className="flex-1" // Not supported by box so fine to add as className
+  >
+    <Text
+      variant={TextVariant.BodyMd}
+      fontWeight={FontWeight.Medium}
+    >
+      Content
+    </Text>
+  </Box>
+  ```
+
+  **React Native:**
+
+  ```tsx
+  // ❌ Don't
+  <View style={StyleSheet.create({ container: { borderRadius: 8, backgroundColor: '#F2F4F6', padding: 16 } }).container}>
+    <Text style={{ color: '#24272A', fontSize: 16, fontWeight: '500' }}>Content</Text>
+  </View>
+
+  // ✅ Do
+  <Box
+    borderRadius={BoxBorderRadius.Lg}
+    backgroundColor={BoxBackgroundColor.BackgroundAlternative}
+    p={4}
+    twClassName="flex-1" // Not supported by box so fine to add as className
+  >
+    <Text
+      variant={TextVariant.BodyMd}
+      fontWeight={FontWeight.Medium}
+    >
+      Content
+    </Text>
+  </Box>
+  ```
+
+### 2. Design Token Integration
+
+- **Use Design Token Generated Classes Only**: Always use design token based classes that are generated from our design system
+- **No Default Tailwind Values**: Never use Tailwind's default color palette or typography scale (these should be removed in the projects `tailwind.config.js` to prevent usage)
+- **Typography**: Always use the `Text` component for text styling instead of Tailwind's typography classes
+- **Colors**: Use only design token generated color classes (e.g., `bg-default`, `text-error-default`)
+- **Example**:
+
+  **React Web:**
+
+  ```tsx
+  // ❌ Don't - Using default Tailwind colors, arbitrary color values or direct typography
+  <div className="bg-[#4459ff] text-white text-sm">
+    <p className="text-gray-900 text-lg font-medium">Title</p>
+    <span className="text-gray-600">Content</span>
+  </div>
```

Comment:
Looking at adding eslint-plugin-tailwind to catch these. We should probably be using our own eslint-plugin-design-tokens which would catch some of these